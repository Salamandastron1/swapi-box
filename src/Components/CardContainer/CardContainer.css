.card-container {
background: transparent;
display: flex;
flex-direction: row;
flex-wrap: wrap;
grid-area: card-container;
grid-gap: 10px;
justify-content: space-around;
overflow: auto;
padding: 10px;
position: fixed;
z-index: 2;
overflow: hidden;
height: 100%;
overflow-y: scroll;

}

.no-favs{
  color: #FEDA4B;
  font-size: 2em;
}