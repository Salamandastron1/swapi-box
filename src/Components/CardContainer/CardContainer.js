import React from 'react';
import './CardContainer.css';
import Card from '../Card/Card';
import * as API from '../API/API';

const CardContainer = ({starWarsDirectory}) => {
  let cards;
  if(typeof starWarsDirectory.people === 'object') {
    cards = starWarsDirectory.people.map((card, i) => {
      return <Card
                key={i + 'people'}
                name={card.name}
                properties={card.properties}
                /> 
              })
            }
    return (
      <section className="card-container">
        {cards}
      </section>
    );
}

export default CardContainer;