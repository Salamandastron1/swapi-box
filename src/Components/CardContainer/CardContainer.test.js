import React from 'react';
import { shallow, mount } from 'enzyme';
import CardContainer from './CardContainer';

describe('CardContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CardContainer />);
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});