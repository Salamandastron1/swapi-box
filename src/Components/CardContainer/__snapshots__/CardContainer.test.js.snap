// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer matches the snapshot 1`] = `
<section
  className="card-container"
>
  <Card />
  <Card />
  <Card />
  <Card />
  <Card />
  <Card />
  <Card />
  <Card />
</section>
`;
